packages/terra-show-hide/src/ShowHideFocuser.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { forwardRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNamesBind from 'classnames/bind';
+import { v4 as uuidv4 } from 'uuid';
+import styles from './ShowHide.module.scss';
+
+const cx = classNamesBind.bind(styles);
+
+const propTypes = {
+  /**
+   * Text that precedes the focusable portion.
+   */
+  focusableText: PropTypes.string.isRequired,
+  /**
+   * The portion of the text that gets focus once the hidden portion is disclosed to be announced by assistive technologies
+   */
+  prefix: PropTypes.string,
+};
+
+/**
+ * @param str - string value
+ * @returns a trimmed string with trailing white space added
+ */
+const addTrailingWhiteSpace = (str) => `${str.trim()} `;
+
+/**
+ * @param str - string value
+ * @returns a trimmed string with preceding whitespace added
+ */
+const addHeadingWhiteSpace = (str) => ` ${str.trim()}`;
+
+/**
+ * This method splits the string in two portions without splitting words. The first (focusable) portion's length should not exceed max length for assistive technologies in the text mode being able to use Arrow Up?Down correctly.
+ * @param string - the text
+ * @returns the array of strings, where the first value is short focusable string and the second value is the rest of the string (if such exists)
+ */
+const truncate = (string) => {
+  if (!string) {
+    return null;
+  }
+  const text = string.trim();
+  const maxLength = 155;
+  if (text?.length < maxLength) {
+    return [text];
+  }
+  const lastFullWordEndPosition = text.slice(0, maxLength).lastIndexOf(' ');
```

Comment:
What happens if the 156th character is the space?

Reply:
It is ok if the 156 is the space. The next two lines here would trim the spaces:
https://github.com/cerner/terra-core/blob/67dcc73138ea81f7db11259eff5e3b952e955937/packages/terra-show-hide/src/ShowHideFocuser.jsx#L47-L48
If the second portion is an empty string it will just add an empty sting at the end of the span with text, which is totally fine:
[      {truncatedText[1] ? addHeadingWhiteSpace(truncatedText[1]) : null}](https://github.com/cerner/terra-core/blob/67dcc73138ea81f7db11259eff5e3b952e955937/packages/terra-show-hide/src/ShowHideFocuser.jsx#L83)

Reply:
Your maxlength is 155.  If the 156th character is a space, it won't find the index of that space right?  Won't it then remove this last word?  I am wondering if this boundary is not handled properly.